### PR TITLE
update cache according to remote changes

### LINKED ISSUE #32: check remote changes of file
Implement a method that checks changes in cached files. 

Ideas:
- when a database is updated, compare it with the current database, if a file is different remove it from cache and fs (file hash is required for comparison) 
- save hash value in cache among with filename. compare it with hash value in the database. invalidate/validate cache accordingly *****

### PR DESCRIPTION
**Pull request by paddlesteamer**
new go-cache

polish

Closes #32

### CODE REVIEW

File: internal/manager/cache.go [hunk 1]
```diff
@@ -54,7 +56,7 @@ type trackerEntry struct {
 
 const idleTimeThreshold time.Duration = 10 * time.Second
 
-func accessFilter(it *cache.Item) bool {
+func accessFilter(key string, it *cache.Item) bool {
```
Comment (utkuufuk) [inline internal/manager/cache.go:59]: How is `key` used in this function?

File: internal/common/utils.go [hunk 2]
```diff
@@ -46,3 +47,13 @@ func ObfuscateFileName(name string) string {
...
+
+func ToString(i int64) string {
+	return strconv.FormatInt(i, 10)
+}
+
+func ToInt64(s string) int64 {
+	i, _ := strconv.ParseInt(s, 10, 0)
+
+	return i
+}
```
Comment (utkuufuk) [inline internal/common/utils.go:59]: Can we not use `Atoi` and `Itoa` instead?
https://golang.org/pkg/strconv/

Comment (paddlesteamer) [reply internal/common/utils.go:59]: Unfortunately no. `Itoa` and `Atoi` works with `int`, not `int64`. The necessity of `int64` goes back to FUSE implementations where the `inode`s are strictly declared as `int64` to be compatible with 32bit systems I think.

File: internal/manager/cache.go (hunk 1 above)
Comment (paddlesteamer) [reply internal/manager/cache.go:59]: Due to the latest change in [go-cache](https://github.com/paddlesteamer/go-cache), filter functions must be of type `filter`: https://github.com/paddlesteamer/go-cache/blob/8516d02047f9c4229e500d1ded7c158b43b78d29/cache.go#L1055

That's why there is an unused parameter. It's the design choice of the `paddlesteamer/go-cache`'s developer (which is me :D) and there's nothing I can do about it haha. 

If you're curious about why the `key` parameter may be necessary, the key is used in inline filter function here: https://github.com/paddlesteamer/cloudstash/blob/c3388f135a5a04f66c32e52fc93bd028f4614a01/internal/manager/background.go#L84-L95

**Pull request merged**